src/promptflow/promptflow/_sdk/_submitter/utils.py

```diff
@@ -198,21 +203,41 @@ def resolve_connections(flow: Flow, client=None, connections_to_ignore=None) ->
         )
 
     @staticmethod
-    def resolve_connection_names_from_tool_meta(tools_meta: dict, flow_dag: dict,):
-        connection_names = set({})
-        tool_names = set({})
-        if tools_meta:
-            packages = tools_meta.get("package", {})
-            for key, pkg in packages.items():
-                inputs = pkg.get("inputs", {})
-                if "connection" in inputs and inputs["connection"].get("type", []) == ["object"]:
-                    tool_names.add(key)
-            nodes = flow_dag.get("nodes", [])
-            for node in nodes:
-                if node.get("source", {}).get("tool", "") in tool_names:
-                    connection_names.add(node["inputs"]["connection"])
-            return list(connection_names)
-        return []
+    def resolve_used_connections(flow: ProtectedFlow, tools_meta: dict, client, connections_to_ignore=None) -> dict:
```

Comment:
Will we have a connection sub module to handle all these connection stuffs? It seems to be very complicated now and such codes are put in several different files.

Reply:
We could do this in future PR.

Reply:
created a task to track this.